cobalt/updater/configurator.cc

```diff
@@ -0,0 +1,422 @@
+      "certscope", GetDeviceProperty(kSbSystemPropertyCertificationScope)));
+
+  // Compression status
+  params.insert(std::make_pair("usecompressedupdates",
+                               GetUseCompressedUpdates() ? "True" : "False"));
+
+  return params;
+}
+
+std::string Configurator::GetDownloadPreference() const {
+  return {};
+}
+
+scoped_refptr<update_client::NetworkFetcherFactory>
+Configurator::GetNetworkFetcherFactory() {
+  return network_fetcher_factory_;
+}
+
+scoped_refptr<update_client::CrxDownloaderFactory>
+Configurator::GetCrxDownloaderFactory() {
+  return crx_downloader_factory_;
+}
+
+scoped_refptr<update_client::UnzipperFactory>
+Configurator::GetUnzipperFactory() {
+  return unzip_factory_;
+}
+
+scoped_refptr<update_client::PatcherFactory> Configurator::GetPatcherFactory() {
+  return patch_factory_;
+}
+
+bool Configurator::EnabledDeltas() const {
+  return false;
+}
+
+bool Configurator::EnabledBackgroundDownloader() const {
+  return false;
+}
+
+bool Configurator::EnabledCupSigning() const {
+  return false;
+}
+
+PrefService* Configurator::GetPrefService() const {
+  return pref_service_.get();
+}
+
+update_client::ActivityDataService* Configurator::GetActivityDataService()
+    const {
+  return nullptr;
+}
+
+bool Configurator::IsPerUserInstall() const {
+  return true;
+}
+
+absl::optional<bool> Configurator::IsMachineExternallyManaged() const {
+  return false;
+}
+
+update_client::UpdaterStateProvider Configurator::GetUpdaterStateProvider()
+    const {
+  return base::BindRepeating([](bool /*is_machine*/) {
+    return update_client::UpdaterStateAttributes();
+  });
+}
+
+std::string Configurator::GetAppGuidHelper(const std::string& updater_channel,
+                                           const std::string& version,
+                                           const int sb_version) {
+  if (updater_channel == "ltsnightly" || updater_channel == "ltsnightlyqa") {
+    return kOmahaCobaltLTSNightlyAppID;
+  }
+  if (version.find(".lts.") == std::string::npos &&
+      version.find(".master.") != std::string::npos) {
+    return kOmahaCobaltTrunkAppID;
+  }
+  std::string channel(updater_channel);
+  // This regex matches to all static channels for C25 and newer in the format
+  // of XXltsY.
+  if (std::regex_match(updater_channel,
+                       std::regex("(2[5-9]|[3-9][0-9])lts\\d+"))) {
+    channel = "static";
+  }
+  auto it = kChannelAndSbVersionToOmahaIdMap.find(channel +
+                                                  std::to_string(sb_version));
+  if (it != kChannelAndSbVersionToOmahaIdMap.end()) {
+    return it->second;
+  }
+  LOG(INFO) << "Configurator::GetAppGuidHelper updater channel and starboard "
+            << "combination is undefined with the new Omaha configs.";
+
+  // All undefined channel requests go to prod configs except for static
+  // channel requestsf for C24 and older.
+  if (!std::regex_match(updater_channel, std::regex("2[0-4]lts\\d+")) &&
```

Comment:
regex use is banned in Chromium: https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-features.md#regex_banned

you'll have to use one of the matchers in 
https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/main:base/substring_set_matcher/ or re2